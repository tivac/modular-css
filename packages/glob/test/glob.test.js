"use strict";

const namer = require("@modular-css/test-utils/namer.js");
const glob = require("../glob.js");

describe("/glob.js", () => {
    it("should be a function", () => {
        expect(typeof glob).toBe("function");
    });

    it("should use a default search", async () => {
        const processor = await glob({
            namer,
            cwd : "./packages/glob/test/specimens",
        });
        
        const { css } = await processor.output();

        expect(css).toMatchSnapshot();
    });

    it("should find files on disk & output css", async () => {
        const processor = await glob({
            namer,
            cwd    : "./packages/glob/test/specimens",
            search : [
                "**/*.css",
            ],
        });
        
        const { css } = await processor.output();

        expect(css).toMatchSnapshot();
    });

    it("should support exclusion patterns", async () => {
        const processor = await glob({
            namer,
            cwd    : "./packages/glob/test/specimens",
            search : [
                "**/*.css",
                "!**/exclude/**",
            ],
        });
        
        const { css } = await processor.output();

        expect(css).toMatchSnapshot();
    });
});
