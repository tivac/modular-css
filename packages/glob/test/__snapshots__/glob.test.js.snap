// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/glob.js should find files on disk & output css 1`] = `
"/* 1.css */
.mc_one {
    color: red;
}
/* dir/3.css */
.mc_three {
    color: white;
}
/* exclude/4.css */
.mc_four {
    color: red;
}
/* dir/2.css */
.mc_two {
    background: blue;
}
"
`;

exports[`/glob.js should support exclusion patterns 1`] = `
"/* 1.css */
.mc_one {
    color: red;
}
/* dir/3.css */
.mc_three {
    color: white;
}
/* dir/2.css */
.mc_two {
    background: blue;
}
"
`;

exports[`/glob.js should use a default search 1`] = `
"/* 1.css */
.mc_one {
    color: red;
}
/* dir/3.css */
.mc_three {
    color: white;
}
/* exclude/4.css */
.mc_four {
    color: red;
}
/* dir/2.css */
.mc_two {
    background: blue;
}
"
`;
